firedrake/matrix_free/operators.py

```diff
@@ -377,11 +377,20 @@ def createSubMatrix(self, mat, row_is, col_is, target=None):
         row_ises = self._y.function_space().dof_dset.field_ises
         col_ises = self._x.function_space().dof_dset.field_ises
 
-        row_inds = find_sub_block(row_is, row_ises, comm=self.comm)
-        if row_is == col_is and row_ises == col_ises:
-            col_inds = row_inds
-        else:
-            col_inds = find_sub_block(col_is, col_ises, comm=self.comm)
+        try:
+            row_inds = find_sub_block(row_is, row_ises, comm=self.comm)
+            if row_is == col_is and row_ises == col_ises:
+                col_inds = row_inds
+            else:
+                col_inds = find_sub_block(col_is, col_ises, comm=self.comm)
+        except LookupError:
+            # Attemping to extract a submatrix that does not match with a subfield
+            # use default PETSc implementation (MatCreateSubMatrixVirtual) via MATSHELL
```

Comment:
I'm assuming that this is two sentences and not one?
```suggestion
            # Attemping to extract a submatrix that does not match with a subfield.
            # Use default PETSc implementation (MatCreateSubMatrixVirtual) via MATSHELL instead.
```